src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -222,6 +222,8 @@ void GcsResourceManager::HandleGetAllResourceUsage(
     reply->mutable_resource_usage_data()->CopyFrom(batch);
   }
 
+  RAY_CHECK_EQ(static_cast<size_t>(reply->resource_usage_data().batch().size()),
+               num_alive_nodes_);
```

Comment:
curious why the size of `node_resource_usages_` not reflecting the number of alive nodes?

Reply:
Because of the bug: when a node is marked as dead, it might still have in-flight resource usage report to gcs and gcs will add it to `node_resource_usages_`